rust/src/operations/transaction/state.rs

```diff
@@ -82,6 +84,45 @@ impl DeltaTableState {
 
         Ok(sql_to_rel.sql_to_expr(sql, &df_schema, &mut Default::default())?)
     }
+
+    /// Get the pysical table schema.
+    ///
+    /// This will construct a schema derived from the parqet schema of the latest data file,
+    /// and fields for partition columns from the schema defined in table meta data.
+    pub async fn physical_arrow_schema(
+        &self,
+        object_store: Arc<dyn ObjectStore>,
+    ) -> DeltaResult<ArrowSchemaRef> {
+        let file_meta = self
+            .files()
+            .iter()
+            .max_by_key(|obj| obj.modification_time)
+            .ok_or(DeltaTableError::Generic("No active file actions to get physical schema. Maybe the current state has not yet been loaded?".into()))?
```

Comment:
Does this error propagate to the user? Does this mean trying to scan an empty tables leads to an error? I don't think it should.

Reply:
It does. at the time I though we could fail on a scan if no files have been added yet, but you are right there are several valid scenarios where we have no files in a table and still should be able to do a scan.

Fixed it so we a re falling back to the schema from metadata.